src/conn/conn_api.c

```diff
@@ -3323,6 +3319,18 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
     /* Start the worker threads, run recovery, and initialize the disaggregated storage. */
     WT_ERR(__wti_connection_workers(session, cfg));
 
+    /*
+     * TODO: WT-15017 we can set this earlier after we have moved the precise checkpoint config
+     * outside the checkpoint server config.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "preserve_prepared", &cval));
```

Comment:
do we have to inforce this condition the other way around as well, so throws error if  precise checkpoint is on but preserve_prepared is off?

Reply:
sure, but should we enforce the config requirement for now?

Reply:
Let's discuss on Friday first.